processor/metricstransformprocessor/README.md

```diff
@@ -17,12 +17,15 @@ The metrics transform processor can be used to rename metrics, labels, or label
   - Aggregation_type: sum, mean, max
 - Add label to an existing metric
 - When adding or updating a label value, specify `{{version}}` to include the application version number
+- Convert resource attributes to metric labels
 
 ## Configuration
 ```yaml
 # transforms is a list of transformations with each element transforming a metric selected by metric name
 transforms:
   # name is used to match with the metric to operate on. This implementation doesn’t utilize the filtermetric’s MatchProperties struct because it doesn’t match well with what I need at this phase. All is needed for this processor at this stage is a single name string that can be used to match with selected metrics. The list of metric names and the match type in the filtermetric’s MatchProperties struct are unnecessary. Also, based on the issue about improving filtering configuration, it seems like this struct is subject to be slightly modified.
+
+  # if the value for `metric_name` is not set (empty string), the operations will be applied to all the metrics. However, currently, it only works and is tested for the 'convert_resource_attributes_to_labels' operation. Existing experience works fine with correct `metric_name`. 
```

Comment:
>it only works and is tested for the 'convert_resource_attributes_to_labels' operation.

This is not what I was expecting. I don't think we want a modification like this. 

We discussed this in the SIG meeting yesterday where @amanbrar1999 and @anuraaga were present. My opinion is that it is the responsibility of exporters to do this transformation. If we want a processor to do this then I would expect it to be a separate specialized processor, provided that for whatever reason we are unable to fix the exporter. I don't see this fitting well in the metricstransformprocessor. The new functionality is not a metric transformation. It is a transformation of a resource into metric labels.

However, ultimately as I said I think this is a bug/deficiency of the Prometheus exporter.

Reply:
```it only works and is tested for the 'convert_resource_attributes_to_labels' operation.```

```**This is not what I was expecting. I don't think we want a modification like this.**```
As I mentioned in the PR description, we created an issue [#1342] to work on that and it does not break existing user experience. I can work on that after this change or maybe with this change if community doesn't agree.

However, I don't clearly get why it should not go with `metrictransform` processor. In a sense, it's adding new labels to the metrics which means transforming the existing metrics. 

Also, I will sync up with @anuraaga and @amanbrar1999.

Reply:
To clarify myself- `metricstransform` processor already has an operation `add_label` and this incoming change is a similar operation which picks the labels from resource attributes. It's a simple new operation type which gives me a feeling to make it a part of this processor.

Also, our main target is `awsemf` exporter not the `prometheus` exporter. This tells us a strong point to keep this in the processor. What if our customer uses a new exporter where we don't have this feature. Then, we are kind of blocking our customers. Having this option in the processor will give our customer to choose any exporters they prefer.

Please correct me if I am wrong. Expecting a second thought @tigrannajaryan .

Reply:
>However, I don't clearly get why it should not go with metrictransform processor. In a sense, it's adding new labels to the metrics which means transforming the existing metrics.

This proposal adds something that does not fit nicely the metrictransform's intent of transforming metrics. This proposal converts resources into metric labels, which is not a metric transformation.